### PR TITLE
Feature: add tag in project grid

### LINKED ISSUE #253: HomePage: filter project by tags(local,remote)
## Description

We want to filter project by tag in the project grid.
Available tags:
- `Local`: display project that don't have repository url
- `remote`: display project that have repository url

Expected feature:
- tag have 2 state active and not active, if active it's colorized. And it can be activated and deactivad by click on it
- multi tag can be activated at the same time
- Expect to have functions to check `isRemoteProject` and `isLocalProject`

Expected design:
image

### PR DESCRIPTION
**Pull request by Zorin95670**
Resolved #253

### CODE REVIEW

File: src/composables/Project.js [hunk 1]
```diff
@@ -46,6 +47,10 @@ function addDateToProjects(projects) {
 export function getProjects() {
   const allProjects = JSON.parse(localStorage.getItem(PROJECT_STORAGE_KEY) || '{}');
 
+  Object.keys(allProjects).forEach((key) => {
```
Comment (k-y0u) [inline src/composables/Project.js:50]: chain .map() to JSON.parse() instead

File: src/components/grid/ProjectGrid.vue [hunk 2]
```diff
@@ -69,13 +73,50 @@
...
+  local: (project) => project.isLocal,
+  remote: (project) => project.isRemote,
+};
+const filteredProjects = computed(() => tags.value
+  .filter(({ active }) => active)
+  .reduce((allProjects, { key }) => allProjects.filter(tagFilters[key]), props.projects)
+  .slice(0, 12));
+
+/**
+ * Toggle tag activation and init project list.
```
Comment (justinehell) [inline src/components/grid/ProjectGrid.vue:99]: init project list where?

File: src/components/grid/ProjectGrid.vue [hunk 3]
```diff
@@ -69,13 +73,50 @@
...
+ * @param {String} key - Tag key
+ * @param {Boolean} isActive - Is tag is activated.
```
Comment (justinehell) [inline src/components/grid/ProjectGrid.vue:101]: If tag is activated.

File: src/components/grid/ProjectGrid.vue [hunk 4]
```diff
@@ -69,13 +73,50 @@
...
+  remote: (project) => project.isRemote,
+};
+const filteredProjects = computed(() => tags.value
+  .filter(({ active }) => active)
+  .reduce((allProjects, { key }) => allProjects.filter(tagFilters[key]), props.projects)
+  .slice(0, 12));
+
+/**
+ * Toggle tag activation and init project list.
+ * @param {String} key - Tag key
```
Comment (justinehell) [inline src/components/grid/ProjectGrid.vue:100]: missing .

File: src/components/grid/ProjectGrid.vue (hunk 2 above)
Comment (Zorin95670) [reply src/components/grid/ProjectGrid.vue:99]: done

File: src/components/grid/ProjectGrid.vue (hunk 4 above)
Comment (Zorin95670) [reply src/components/grid/ProjectGrid.vue:100]: done

File: tests/unit/models/Project.spec.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import Project from 'src/models/Project';
+
+describe('Test class: Project', () => {
+  describe('Test constructor', () => {
+    it('should use default values when called with no parameter', () => {
```
Comment (justinehell) [inline tests/unit/models/Project.spec.js:5]: `with no` -> `without`

File: src/i18n/en-US/index.js [hunk 6]
```diff
@@ -407,4 +407,8 @@ export default {
       },
     },
   },
+  tag: {
```
Comment (justinehell) [inline src/i18n/en-US/index.js:410]: tags?

File: cypress/e2e/HomePage/filter.project.feature [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    # Go back to HomePage
+    When I click on '[data-cy="navigation-bar"] [data-cy="home-page-link"]'
+    Then I expect current url is '/'
+
+    # Verify all project tags are inactive
+    And  I expect '[data-cy="inactive-tag_local"]' exists
+    And  I expect '[data-cy="inactive-tag_remote"]' exists
+
+    # Select local tag and verify only local projects are display
```
Comment (justinehell) [inline cypress/e2e/HomePage/filter.project.feature:37]: displayed

File: cypress/e2e/HomePage/filter.project.feature [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    And  I expect '[data-cy="project-card_{{localProjectName}}"]' appear 1 time on screen
+
+    # Select remote tag and verify to have no project present
+    When I click on '[data-cy="inactive-tag_remote"]'
+    Then I expect '[data-cy="inactive-tag_remote"]' not exists
+    And  I expect '[data-cy="active-tag_remote"]' exists
+    And  I expect '[data-cy="project-card_{{remoteProjectName}}"]' not exists
+    And  I expect '[data-cy="project-card_{{localProjectName}}"]' not exists
+
+    # Unselect local tag and verify only remote projects are display
```
Comment (justinehell) [inline cypress/e2e/HomePage/filter.project.feature:51]: displayed

File: cypress/e2e/HomePage/filter.project.feature [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    And  I expect '[data-cy="inactive-tag_remote"]' exists
+
+    # Select local tag and verify only local projects are display
+    When I click on '[data-cy="inactive-tag_local"]'
+    Then I expect '[data-cy="inactive-tag_local"]' not exists
+    And  I expect '[data-cy="active-tag_local"]' exists
+    And  I expect '[data-cy="project-card_{{remoteProjectName}}"]' not exists
+    And  I expect '[data-cy="project-card_{{localProjectName}}"]' appear 1 time on screen
+
+    # Select remote tag and verify to have no project present
```
Comment (justinehell) [inline cypress/e2e/HomePage/filter.project.feature:44]: Select the remote tag and check that no project is present

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/i18n/en-US/index.js (hunk 6 above)
Comment (Zorin95670) [reply src/i18n/en-US/index.js:410]: all other key are in singualr

File: src/models/Project.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+ * Representing a user project.
+ */
+class Project {
+  /**
+   * Default constructor.
+   * @param {Object} [props={}] - Object that contains all properties to set.
+   * @param {String} [props.id=null] - The id of this project.
+   * @param {Number} [props.creationDate=null] - The creation date of this project in timestamp.
+   * @param {Object} [props.git={}] - Git settings of this project.
+   * @param {String} [props.git.token=null] - Access token of repository.
```
Comment (k-y0u) [inline src/models/Project.js:11]: missing "of this project"

File: tests/unit/models/Project.spec.js (hunk 5 above)
Comment (Zorin95670) [reply tests/unit/models/Project.spec.js:5]: done

File: cypress/e2e/HomePage/filter.project.feature (hunk 7 above)
Comment (Zorin95670) [reply cypress/e2e/HomePage/filter.project.feature:37]: done

File: cypress/e2e/HomePage/filter.project.feature (hunk 8 above)
Comment (Zorin95670) [reply cypress/e2e/HomePage/filter.project.feature:51]: done

File: cypress/e2e/HomePage/filter.project.feature (hunk 9 above)
Comment (Zorin95670) [reply cypress/e2e/HomePage/filter.project.feature:44]: done

File: src/models/Project.js (hunk 10 above)
Comment (Zorin95670) [reply src/models/Project.js:11]: done

File: cypress/e2e/HomePage/filter.project.feature [hunk 11]
```diff
@@ -0,0 +1,56 @@
+Feature: Test home page: project filter
+
+  Scenario: Select/unselect tag should filter the project list
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/HomePage/filter.project.feature:3]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**